server/tests/util/test_binpacking.py

```diff
@@ -242,6 +349,67 @@ def test_balance(self, bucketlist):
             batches == new_batches
         ), "Balanced batches were not the same as original batches!"
 
+    def test_balance_skewed(self, skewedbucketlist):
+        new_bl = skewedbucketlist.balance()
+
+        assert (
+            skewedbucketlist.deviation() >= new_bl.deviation()
+        ), "Balanced list has higher deviation than original assignment! {} is greater than {}".format(
+            new_bl.deviation(), skewedbucketlist.deviation()
+        )
+
+        num_batches = sum([len(bucket.batches) for bucket in skewedbucketlist.buckets])
```

Comment:
fun fact: you don't need to pass `sum` a list, it can take an iterator. i.e. you can omit the square brackets

Reply:
cool!